### PR TITLE
Refactored distribution conf into json

### PR DESCRIPTION
**Pull request by breillyr**

### CODE REVIEW

File: dockpulp/__init__.py [hunk 1]
```diff
@@ -1021,25 +1015,38 @@ def createRepo(self, repo_id, url, registry_id=None, desc=None, title=None, prot
...
+                raise errors.DockPulpConfigError("Distribution %s not defined in %s" %
+                                                 (distribution, DEFAULT_DISTRIBUTIONS_FILE))
             if productline:
-                if not productline.endswith(self.name_enforce.get(distribution, '')):
-                    raise errors.DockPulpError("%s is a %s repo, product-line must end with %s" %
+                if not productline.endswith(distconf.get('name_enforce', '')):
+                    raise errors.DockPulpError("%s is a %s repo, product-line must end with -%s" %
                                                (repo_id, distribution, distribution))
+                for restrict in distconf['name_restrict']:
+                    if productline.find(restrict) != -1:
```
Comment (lcarva) [inline dockpulp/__init__.py]: Would it be cleaner to simply: `if restrict in productline:` ?

File: dockpulp/__init__.py [hunk 2]
```diff
@@ -1021,25 +1015,38 @@ def createRepo(self, repo_id, url, registry_id=None, desc=None, title=None, prot
...
+                            "%s is a %s repo, product-line must not contain %s" %
+                            (repo_id, distribution, restrict))
             # library level repo do not use product-line
             elif library:
-                if not repo_id.endswith(self.name_enforce.get(distribution, '')):
+                if not repo_id.endswith(distconf.get('name_enforce', '')):
                     raise errors.DockPulpError("%s is a %s repo, repo id must end with %s" %
                                                (repo_id, distribution, distribution))
+                for restrict in distconf['name_restrict']:
+                    if repo_id.find(restrict) != -1:
```
Comment (lcarva) [inline dockpulp/__init__.py]: Would it be cleaner to simply: `if restrict in repo_id:` ?

File: dockpulp/__init__.py [hunk 3]
```diff
@@ -1021,25 +1015,38 @@ def createRepo(self, repo_id, url, registry_id=None, desc=None, title=None, prot
...
+            try:
+                distconf = self.distributionconf[distribution]
+            except KeyError:
+                raise errors.DockPulpConfigError("Distribution %s not defined in %s" %
+                                                 (distribution, DEFAULT_DISTRIBUTIONS_FILE))
             if productline:
-                if not productline.endswith(self.name_enforce.get(distribution, '')):
-                    raise errors.DockPulpError("%s is a %s repo, product-line must end with %s" %
+                if not productline.endswith(distconf.get('name_enforce', '')):
+                    raise errors.DockPulpError("%s is a %s repo, product-line must end with -%s" %
```
Comment (lcarva) [inline dockpulp/__init__.py]: Where's the dash, `-`, being enforced? Should the check be if productline endswith `'-' + name_enforce`? Or should the dash be removed from error message?

Comment (breillyr) [inline dockpulp/__init__.py]: name_enforce is now a string, so the dash can be explicitly defined in the json

Comment (breillyr) [inline dockpulp/__init__.py]: Ah but I see your point now :) Will remove it from the error message...

**Pull request merged**